BuildScript/BuildTargets/Testing.targets

```diff
@@ -63,53 +63,38 @@
       <RunTestsStartTime>$([System.DateTime]::UtcNow)</RunTestsStartTime>
     </PropertyGroup>
   </Target>
-  
-  <!-- 
-    Actual testing happens here.
-    BuildTestOutputFilesCrossProduct creates all test configurations 
-    (the result is a cross product of TestOutputFiles x AllPlatforms x AllDatabaseSystems x AllBrowsers), 
-    see description on the target for more information. Afterwards _testOutputFiles contains entries like:
-    
-      Identity = MyUnitTests1.dll ; Platform = x86 ; DatabaseSystem = SqlServer2005 ; Browser = NoBrowserRequired
-      Identity = MyUnitTests1.dll ; Platform = x64 ; DatabaseSystem = SqlServer2005 ; Browser = NoBrowserRequired
-      Identity = MyUnitTests1.dll ; Platform = x86 ; DatabaseSystem = SqlServer2008 ; Browser = NoBrowserRequired
-      ...
-      Identity = MyUnitTests2.dll ; Platform = x86 ; DatabaseSystem = NoDatabaseRequired ; Browser = NoBrowserRequired
-      ...
-      Identity = MyWebTests1.dll ; Platform = x86 ; DatabaseSystem = NoDatabaseRequired ; Browser = Chrome
-      ...
-      (only relevant metadata shown)
-      
-    RunTestsInternal uses target batching to be called once for each combination of "Identity;Platform;DatabaseSystem;Browser".
-    By batching over all the parameters, the _testOutputFiles-ItemGroup contains only a single item per batch.
-  -->
-  <Target Name="RunTestsInternal" DependsOnTargets="AddAdditionalMetadataToOutputFiles;BuildTestOutputFilesCrossProduct;PrepareNunitDirectory" Outputs="%(_testOutputFiles.Identity);%(_testOutputFiles.Platform);%(_testOutputFiles.DatabaseSystem);%(_testOutputFiles.Browser)">
-    <Message Text="Running tests %(_testOutputFiles.FullPath), ConfigurationID=$(ConfigurationID), Additional Configuration Information = %(_testOutputFiles.Platform), %(_testOutputFiles.DatabaseSystem), %(_testOutputFiles.Browser)" Importance="High"/>
-
-    <Error Text="The property 'MSBuildExtensionPackPath' is not set." Condition="'$(MSBuildExtensionPackPath)' == ''" />
-    <Error Text="The property 'LogDirectory' is not set." Condition="'$(LogDirectory)' == ''" />
-    <Error Text="The property 'NoDatabaseRequired' is not set." Condition="'$(NoDatabaseRequired)' == ''" />
-    <Error Text="The property 'NoBrowserRequired' is not set." Condition="'$(NoBrowserRequired)' == ''" />
-    <Error Text="The property 'NUnitToolPath' is not set." Condition="'$(NUnitToolPath)' == ''" />
-    <Error Text="The property 'NunitFolderPathWithDefaultConfiguration' is not set." Condition="'$(NunitFolderPathWithDefaultConfiguration)' == ''" />
-    <Error Text="The property 'NunitFolderPathWithoutDotNet35Dependency' is not set." Condition="'$(NunitFolderPathWithoutDotNet35Dependency)' == ''" />
+
+
+  <Target Name="RunTestsInternal" DependsOnTargets="AddAdditionalMetadataToOutputFiles;CreateTestConfigurations;PrepareNunitDirectory" Outputs="%(_testOutputFiles.Identity);%(_testOutputFiles.Platform);%(_testOutputFiles.DatabaseSystem);%(_testOutputFiles.Browser)">
```

Comment:
linewrap the attributes

Reply:
Do we not need output batching for the other parameters, too?

Reply:
What about using the identity spec or a new guid value as a single discriminator value that is unique accross al permuations? That way, we can shorten the batching from multiple metadata to a single item.